### PR TITLE
fix: tests routes

### PR DESCRIPTION
**Pull request by nbeteram**
- [x] Unit tests for `/backend/routes/admin.ts`
- [x] Unit tests for `/backend/routes/coach.ts`
- [ ] Unit tests for `/backend/routes/form.ts`
- [x] Unit tests for `/backend/routes/login.ts`
- [ ] Unit tests for `/backend/routes/project.ts`
- [ ] Unit tests for `/backend/routes/student.ts`
- [x] Unit tests for `/backend/routes/followup.ts`
- [ ] Unit tests for `/backend/routes/template.ts`
(See #199 )

The tests work but the reviewers have to test the frontend:
- can't change your own status
- can't remove yourself
- can't send requests inside manager users if you aren't an admin anymore

### CODE REVIEW

**Review by jay-tux — APPROVED** [review]
🚀 great work @nbeteram

File: backend/request.ts [hunk 1]
```diff
@@ -409,17 +422,23 @@ export async function parseFilterUsersRequest(
...
+        ("statusFilter" in req.body &&
+            req.body.statusFilter !== "ACTIVATED" &&
+            req.body.statusFilter !== "PENDING" &&
+            req.body.statusFilter !== "DISABLED") ||
+        ("isCoachFilter" in req.body &&
+            req.body.isCoachFilter.toString() !== "true" &&
+            req.body.isCoachFilter.toString() !== "false") ||
+        ("isAdminFilter" in req.body &&
+            req.body.isAdminFilter.toString() !== "true" &&
+            req.body.isAdminFilter.toString() !== "false")
```
Comment (BramDevlaminck) [inline backend/request.ts:441]: we could define our own kind of StringBoolean to check if this is valid

File: backend/request.ts [hunk 2]
```diff
@@ -409,17 +422,23 @@ export async function parseFilterUsersRequest(
...
     }
     if (
-        "statusFilter" in req.body &&
-        req.body.statusFilter !== "ACTIVATED" &&
-        req.body.statusFilter !== "PENDING" &&
-        req.body.statusFilter !== "DISABLED"
+        ("statusFilter" in req.body &&
+            req.body.statusFilter !== "ACTIVATED" &&
+            req.body.statusFilter !== "PENDING" &&
+            req.body.statusFilter !== "DISABLED") ||
```
Comment (BramDevlaminck) [inline backend/request.ts:435]: also possible to use `in` here?

File: backend/request.ts [hunk 3]
```diff
@@ -581,15 +600,25 @@ export async function parseFilterProjectsRequest(
...
-                .map((num: string) => Number(num));
+                .map((num: string) => parseInt(num));
         }
     }
 
     let fullyAssignedFilter = maybe(req.body, "fullyAssignedFilter");
     if ("fullyAssignedFilter" in req.body) {
+        if (
+            req.body.fullyAssignedFilter.toString() !== "true" &&
+            req.body.fullyAssignedFilter.toString() !== "false"
```
Comment (BramDevlaminck) [inline backend/request.ts:618]: use the "stringBoolean" type is described earlier

**Review by BramDevlaminck — CHANGES_REQUESTED** [review]
good work! just change those types :)

File: backend/request.ts (hunk 1 above)
Comment (Mouwrice) [reply backend/request.ts:441]: huh, is the `isCoachFilter` not a boolean?

File: backend/request.ts [hunk 4]
```diff
@@ -731,11 +760,12 @@ export async function parseFormRequest(
 export async function parseRequestResetRequest(
     req: express.Request
 ): Promise<Requests.ReqReset> {
-    return hasFields(req, ["email"], types.neither).then(() =>
-        Promise.resolve({
+    return hasFields(req, ["email"], types.neither).then(() => {
+        console.log(validator.default.normalizeEmail("no").toString());
```
Comment (Mouwrice) [inline backend/request.ts:764]: I think we can remove this log statement.

File: backend/request.ts (hunk 1 above)
Comment (Mouwrice) [reply backend/request.ts:441]: There currently is no way to set these filters to false in the frontend. So we might not even need to check if they are true or false.

**Review by Mouwrice — CHANGES_REQUESTED** [review]
Getting there!

**Review by BramDevlaminck — APPROVED** [review]
nice!

**Review by Mouwrice — APPROVED** [review]
Pretty, pretty good

**Pull request merged**